### ISSUE TITLE
Initialization of conjugate gradient in competitive_optimizer 

### ISSUE THREAD

**Issue by f-t-s** [HyperGAN/HyperGAN#255]
Hi,

I saw that you have implemented competitive gradient descent in your dev-branch, thanks for giving it a try @mikkel ! 
I just wanted to mention that you seem to have a bug that we also found in our pytorch implementation recently, and will correct shortly.

In the definition of ``mgeneral_conjugate_gradient`` you initialize the residual with 
```
r = [tf.identity(_b) for _b in b]
```

This is correct if the initial guess is zero, otherwise (if A is the matrix to be inverted), it should be 
r = b - Ax_0 

where x_0 is the initial guess, see also [here](https://en.wikipedia.org/wiki/Conjugate_gradient_method)

Initializing the residual wrongly can lead to different results and we found that there are cases where this bug breaks cgd, while the fixed version performs well.

**Comment by martyn** [comment]
Sorry for missing this. Conjugate gradient is awesome, thank you for your work! We've rewritten hypergan in pytorch and are working to re-add conjugate gradient.

**Comment by martyn** [comment]
Thank you for identifying the nonzero-warm-start case: initializing the residual as `b` only applies when the initial guess is zero; the reported case requires `b - A x0`. The TensorFlow `competitive_optimizer` that contained this code has now been archived and removed from the supported tree in #300. The new ParticleGAN reference loop uses Adam and does not currently implement competitive/conjugate-gradient optimization.

Closing this as superseded by that retirement, **not** as a claim that CGD was fixed or revalidated. If CGD is admitted again, the research acceptance requirements must include a nonzero-initial-guess residual test, comparison against a known solve, and convergence/regression evidence. Historical code remains accessible through the [migration/archive notes](https://github.com/HyperGAN/HyperGAN/blob/develop/docs/migration.md).

Audit record: [individual disposition and retained work](https://github.com/HyperGAN/HyperGAN/blob/f4053ae1/reports/issue-audit-2026-09-18.md), published in #302.
